commons/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java

```diff
@@ -45,23 +48,54 @@
 
    static {
       // Classes always allowed
+      // Primitives
+      SYS_ALLOWED_CLASSES.add(byte.class.getName());
```

Comment:
I don't think a user can put a `byte` or any of the regular primitives into the cache. Only the boxed variants afaik.

Reply:
The `ClassWhiteList` is also used by the transcoders, so I think we will probably need primitives in this context at some point. But either way, I don't see it being an issue including them just in case.

Reply:
Will has a point, it's impossible to use the whitelist for primitives: when you call `writeObject(0.0)` you can't avoid boxing, and the whitelist doesn't apply to `writeDouble(double)`.
If some transcoders get fancy and use the whitelist for primitives too, it's their problem :)

Reply:
Oh actually I just remember it is probably an issue for a class holding those primitives. Do we recursively check inner objects?

Reply:
Inner classes are checked, however primitives aren't, so we don't need to add them for that reason. I'm not 100% on the details, but I think this is because the way `ObjectInputStream` calls `CheckedInputStream::resolveClass`

Reply:
Yeah, all inner objects are checked, otherwise it wouldn't be much of a whitelist: an attacker can always construct a stream so that a whitelisted object contains a non-whitelisted object, and by the time the non-whitelisted object is cast to the field type it's too late.